### DISCUSSION TITLE
How to get the unfiltered count matrix?

### DISCUSSION THREAD

**Discussion by RyanZhiNie** [COMBINE-lab/simpleaf#119] [Q&A] [answered]
Hi, simpleaf team! First of all, thanks for your excellent work in this tool.
I'm here to ask how to get the unfiltered count matrix after running a complete simpleaf scRNA-seq processing workflow. If I didn't misunderstand, the alevin-fry in simpleaf removes inferred empty droplets after it constructs cell barcode permitlists. My purpose is to use the unfiltered count matrix as the input of SoupX, a tool to correct ambient RNA, which estimates the ambient mRNA expression profile from empty droplets in the unfiltered count matrix.
Thanks for your help in advance.

**Comment by rob-p** [comment, marked as answer]
Hi @RyanZhiNie,

  Can you tell us a little bit more about the data type you're using?  If you have a technology with an "external whitelist" (like 10x), then it's very easy to get an unfiltered matrix by just passing the correct flag to `simpleaf quant` for the filter mode.  Specifically:

```
  -u, --unfiltered-pl [<UNFILTERED_PL>]  use unfiltered permit list
``` 

you pass the permit list and it will use this to generate an unfiltered count matrix.

  However, if you're working with a technology that doesn't have an external permit list, this is a bit more difficult.  In that case, you could use the `--expect-cells` option with a largeish number of cells

```
  -e, --expect-cells <EXPECT_CELLS>      use expected number of cells
```

or similarly the `--force-cells` option with a large number of cells

```
  -f, --forced-cells <FORCED_CELLS>      use forced number of cells
```

Best,
Rob

**Reply by RyanZhiNie** [reply]
Thank you for your rapid reply, @rob-p 

I have already processed a Drop-seq dataset with simpleaf. However, because there are no explicit cell barcode permit lists for Drop-seq (maybe there are but I temporarly don't find), I used knee filtering mode with the `-k` option, which may be the reason why empty droplets were removed from the final count matrix. I'll try your suggestion with the `--expect-cells` option.

BTW, I also have some 10x 3' datasets to process and I'll use the `--unfiltered-pl` option. I notice that there is a `--min-reads` option to be used with the `--unfiltered-pl` option, and it defaults to `10`. Unluckily, I don't find a similar option in the [Cell Ranger count](https://www.10xgenomics.com/support/software/cell-ranger/advanced/cr-command-line-arguments) command. So, to get an unfiltered count matrix like Cell Ranger, maybe it is a good way to set it to `0`? Any suggestion?

**Reply by rob-p** [reply]
Hi @RyanZhiNie,

  Sounds good!

>BTW, I also have some 10x 3' datasets to process and I'll use the --unfiltered-pl option. I notice that there is a --min-reads option to be used with the --unfiltered-pl option, and it defaults to 10. Unluckily, I don't find a similar option in the [Cell Ranger count](https://www.10xgenomics.com/support/software/cell-ranger/advanced/cr-command-line-arguments) command. So, to get an unfiltered count matrix like Cell Ranger, maybe it is a good way to set it to 0? Any suggestion?

In this case, I think leaving the `--min-reads` at 10 is OK.  This has come up before with folks who use Cell Bender for filtering and QC, which generally wants as much information as possible to decide. However, the developers there have chimed in that considering a very small minimum read count (like 10) for cells is OK.  In fact, with a small min read (but greater than 1) you generally recover *more* cells, because this allows the barcode correction algorithm to work better than when you set `--min-reads 1`.  This is because in this case `simpleaf` can take advantage of information about the observed abundance of cell barcode to correct barcodes that are otherwise ambiguous (i.e. barcode that are the same edit distance from 2 or more barcodes in the whitelist).

**Reply by RyanZhiNie** [reply]
Hi @rob-p 

Got it. Thanks a lot for all your suggestions!

Hope you have a good day.

Best regards,
Zhi

**Reply by rob-p** [reply]
You're welcome! Have a great day :).
